packages/react-accordion/src/components/Accordion/useAccordion.ts

```diff
@@ -1,38 +1,98 @@
 import * as React from 'react';
-import { resolveShorthand } from '@fluentui/react-utilities';
-import { AccordionProps, AccordionSlots, AccordionState } from './Accordion.types';
-import { useCreateAccordionContextValue } from './useAccordionContext';
+import { useControllableValue, useDescendantsInit, useEventCallback } from '@fluentui/react-utilities';
+import { AccordionDescendant, AccordionIndex, AccordionProps, AccordionSlots, AccordionState } from './Accordion.types';
 
 /**
  * Const listing which props are shorthand props.
  */
 export const accordionShorthandProps: Array<keyof AccordionSlots> = [];
 
-/**
- * Returns the props and state required to render the component
- * @param props - Accordion properties
- * @param ref - reference to root HTMLElement of Accordion
- * @param defaultProps - default values for the properties of Accordion
- */
-export const useAccordion = (props: AccordionProps, ref: React.Ref<HTMLElement>): AccordionState => {
-  const initialState = {
-    ref,
-    collapsible: false,
-    multiple: false,
-    navigable: false,
-    ...props,
-    components: {
-      root: 'div',
-    },
-    button: props.button ? resolveShorthand(props.button) : undefined,
-    expandIcon: props.expandIcon ? resolveShorthand(props.expandIcon) : undefined,
-    icon: props.icon ? resolveShorthand(props.icon) : undefined,
-  } as const;
-  const [context, descendants, setDescendants] = useCreateAccordionContextValue(initialState);
+export const useAccordion = (
+  { index, defaultIndex, multiple = false, collapsible = false, onToggle, navigable = false, ...rest }: AccordionProps,
+  ref: React.Ref<HTMLElement>,
+): AccordionState => {
+  const [descendants, setDescendants] = useDescendantsInit<AccordionDescendant>();
+  const normalizedIndex = React.useMemo(() => normalizeIndex(index), [index]);
+
+  // TODO: fix typings in useControllableValue
+  const [openItems, setOpenItems] = useControllableValue<number[], HTMLElement>(normalizedIndex!, () =>
+    initializeUncontrolledOpenItems({ collapsible, defaultIndex, multiple }),
+  );
+
+  const requestToggle = useEventCallback((ev: React.MouseEvent<HTMLElement>, i: number) => {
+    if (descendants[i]?.disabled === true) {
+      return;
+    }
+    onToggle?.(ev, i);
```

Comment:
Should we make this consistent with how other components are structure the data in the change callbacks ?

```suggestion
    onToggle?.(ev, { item: i });
```